Patterns/French/French-DateTime.yaml

```diff
@@ -1055,6 +1055,11 @@ AmbiguityFiltersDict: !dictionary
   entries:
     '^([eé]t[eé])$': '(?<!((l\s*[''`]\s*)|(cet(te)?|en)\s+))[eé]t[eé]\b'
     '^(mer)$': '(?<!((le|ce)\s+))mer\b'
+# Used to exclude duration matches (e.g. "pour 3 heures") from time extractions
+AmbiguityTimeFiltersDict: !dictionary
+  types: [ string, string ]
+  entries:
+    'heures?$': '(pour|durée\s+de)\s*'
```

Comment:
Shouldn't the second regex check for the <number heures> part too? Otherwise if there are multiple "heures".
Similar to the general filter a few lines above.

Reply:
It does not seem necessary given the current implementation of the filter because it excludes only the extractions ending with "heures" that are immediately preceded by "(pour|durée\s+de)"

Reply:
As discussed in the code file, the current approach there is too brittle. Either we need a way to represent the "contiguity" as part of the filter definition, or we need a filter regex that checks the key too.
Maybe something like (pour|durée\s+de)\s*(\S+){1,2}\sheures?

Reply:
updated regex as suggested and reverted FilterAmbiguity to its original form (from BaseMergedDateTimeExtractor) which checks for overlapping matches.